internal/isoeditor/rhcos.go

```diff
@@ -92,6 +100,65 @@ func (e *rhcosEditor) addIgnitionArchive(ignition string) error {
 	return ioutil.WriteFile(e.isoHandler.ExtractedPath("images/ignition.img"), archiveBytes, 0644)
 }
 
+func (e *rhcosEditor) addCustomRAMDisk(staticIPConfig string) error {
+	configPath := "/etc/static_ips_config.csv"
+	scriptPath := "/usr/lib/dracut/hooks/initqueue/settled/90-assisted-static-ip-config.sh"
+	scriptContent := constants.ConfigStaticIpsScript
+	imagePath := "/images/assisted_installer_custom.img"
+
+	f, err := os.Create(e.isoHandler.ExtractedPath(imagePath))
+	if err != nil {
+		return err
+	}
+
+	w := cpio.NewWriter(f)
+	if err = addFileToArchive(w, configPath, staticIPConfig, 0o664); err != nil {
+		return err
+	}
+	if err = addFileToArchive(w, scriptPath, scriptContent, 0o755); err != nil {
+		return err
+	}
+	if err = w.Close(); err != nil {
+		return err
+	}
+
+	// edit config to add new image to initrd
+	err = editFile(e.isoHandler.ExtractedPath("EFI/redhat/grub.cfg"), `(?m)^(\s+initrd) (.+| )+$`, fmt.Sprintf("$1 $2 %s", imagePath))
```

Comment:
Just wondering if we should do the cfg files modifications in the template. I mean, we could add the ram disk as part of the template, which would simply be empty in the non-static-ip flow. That way, it would be simpler for streaming, as we would need to change only the ramdisk image. But can be tackled later on of course...

Reply:
Yeah, I think we can tackle this for the streaming epic as there's no reason to add an empty image now.

Reply:
Ack